static/app/components/structuredEventData/index.tsx

```diff
@@ -29,73 +75,85 @@ interface StructuredDataProps {
 
 export function StructuredData({
   config,
-  value = null,
+  forceDefaultExpand,
+  initialExpandedPaths,
   maxDefaultDepth,
-  withAnnotatedText,
-  withOnlyFormattedText = false,
   meta,
   objectKey,
-  forceDefaultExpand,
+  onToggleExpand,
+  value = null,
+  withAnnotatedText,
+  withOnlyFormattedText = false,
 }: StructuredDataProps) {
+  const getInitialExpandedPaths = useCallback(() => {
+    return (
+      initialExpandedPaths ??
+      (forceDefaultExpand === false ||
+      (forceDefaultExpand === undefined && !maxDefaultDepth)
+        ? []
+        : getDefaultExpanded(Math.max(1, maxDefaultDepth), value))
+    );
```

Comment:
I didn't write unit tests for this, but the "Auto-Expanded items" story renders the same as before. See https://github.com/getsentry/sentry/pull/74481 as a reference.

Reply:
Should we write a test for this? Seems like a lot of extra logic that could regress

Reply:
tested!